website/registrations/models.py

```diff
@@ -407,6 +408,19 @@ def save(
     ):
         if self.pk is None:
             self.status = Entry.STATUS_REVIEW
+
+        if (
+            self.contribution is None
+            and self.length == Entry.MEMBERSHIP_STUDY
+            and self.member.current_membership is not None
+            and self.member.current_membership.length - Entry.MEMBERSHIP_YEAR
```

Comment:
`self.member.current_membership.length - Entry.MEMBERSHIP_YEAR`

is not a boolean check? if this means `!= 0`, please write this explicitely

Reply:
This was the previous check that we had: https://github.com/svthalia/concrexit/blob/ca6abb32946c961fa7d52bc04b1cd34bfb73e88e/website/registrations/services.py#L254


I believe `Entry.MEMBERSHIP_YEAR` is a string.